frontend/src/scenes/insights/Insight.tsx

```diff
@@ -69,6 +68,18 @@ export function Insight({ insightId }: { insightId: InsightShortId }): JSX.Eleme
         reportHotkeyNavigation('insights', hotkey)
     }
 
+    useEffect(() => {
+        const beforeUnloadHandler = (e: BeforeUnloadEvent): void => {
+            // Cancel the event to show unsaved changes dialog
+            e.preventDefault()
+            e.returnValue = ''
+        }
```

Comment:
This doesn't seem to prompt for anything? Couldn't get this to do anything when testing either.

Reply:
Do you mean you don't get the native browser prompt? 🤔 What browser?